doc/specs/stdlib_linalg.md

```diff
@@ -101,30 +101,57 @@ end program demo_diag5
 
 Experimental
 
+### Class
+
+Pure function.
+
 ### Description
 
-Construct the identity matrix
+Construct the identity matrix.
 
 ### Syntax
 
-`I = [[stdlib_linalg(module):eye(function)]](n)`
+`I = [[stdlib_linalg(module):eye(function)]](dim1 [, dim2])`
 
 ### Arguments
 
-`n`: Shall be a scalar of default type `integer`. 
+`dim1`: Shall be a scalar of default type `integer`.
+This is an `intent(in)` argument. 
+
+`dim2`: Shall be a scalar of default type `integer`.
+This is an `intent(in)` and `optional` argument. 
 
 ### Return value
 
-Returns the identity matrix, i.e. a square matrix with ones on the main diagonal and zeros elsewhere. The return value is of type `integer(int8)`.
+Return the identity matrix, i.e. a matrix with ones on the main diagonal and zeros elsewhere. The return value is of type `integer`.
```

Comment:
Why did you change the return value from `integer(int8)` to `integer`?

Reply:
Is there any specific difference between `integer(int8)` and `integer`? I may think `integer` is a bit more common, but there is no difference in actual use.🤣

Reply:
The default integer typically uses 32-bits of storage. The use of int8 was suggested by Milan, to save storage I believe. Personally I'm fine with using the default integer.

Reply:
Yes, it was mainly to save storage. Since the result would be mainly temporary, and most likely converted, I am in favor to keep it as `integer(int8)`

Reply:
Agree, I will change it. (Also `zeros/ones/empty`)